include/zephyr/net/ieee802154.h

```diff
@@ -26,6 +26,8 @@ extern "C" {
 
 /**
  * @defgroup ieee802154 IEEE 802.15.4 and Thread APIs
+ * @since 1.0
+ * @version 0.8.0
```

Comment:
According to https://docs.zephyrproject.org/latest/develop/api/overview.html, the current version of all IEEE 802.15.4 APIs is now (since release 3.5) `1.0.0` - also concerns the other `ieee802154*` files. Last time I reviewed this was not the case but now I think this should be adjusted.

Reply:
the API based on the latest docs in https://docs.zephyrproject.org/latest/develop/api/overview.html is 'unstable'. 0.8.0 means 'unstable', it should not be 1.0.0 until the API is stable.

@edersondisouza  please put some documentation at the top of the page explaining the versions and what they mean.

Reply:
> According to https://docs.zephyrproject.org/latest/develop/api/overview.html, the current version of all IEEE 802.15.4 APIs is now (since release 3.5) `1.0.0`

FWIW the last column in the table is meant to indicate the API was **introduced** in Zephyr 1.0.0, so that recent update to the table you're mentioning was probably bogus in the first place.